modules/internal/BytesStringCommon.chpl

```diff
@@ -276,6 +281,7 @@ module BytesStringCommon {
     x.buff = other;
     x.buffSize = size;
     x.buffLen = length;
+    if t == string then x.cachedNumCodepoints = x.numCodepoints;
```

Comment:
I don't think we need this. `WithOwnedBuffer` variations of factory functions can't be called with other `string`s but only with C buffers. When that happens, we must be calling `validateEncoding` in the call chain somewhere else.